src/core/ext/filters/client_channel/lb_policy/weighted_target/weighted_target.cc

```diff
@@ -444,44 +445,32 @@ void WeightedTargetLb::UpdateStateLocked() {
 WeightedTargetLb::WeightedChild::DelayedRemovalTimer::DelayedRemovalTimer(
     RefCountedPtr<WeightedTargetLb::WeightedChild> weighted_child)
     : weighted_child_(std::move(weighted_child)) {
-  GRPC_CLOSURE_INIT(&on_timer_, OnTimer, this, nullptr);
-  Ref().release();
-  grpc_timer_init(&timer_, ExecCtx::Get()->Now() + kChildRetentionInterval,
-                  &on_timer_);
+  timer_handle_ = GetDefaultEventEngine()->RunAt(
+      absl::Now() + kChildRetentionInterval, [self = Ref()] {
```

Comment:
Doesn't this lambda need to be `mutable` in order to do the `std::move()` into the nested lambda below?

Reply:
Yes, you're right. I've replaced it with `self = Ref()`. Even though the function will only be called once, making the arg mutable feels like the wrong choice.

Reply:
Done. As discussed, the fact that [this behavior](https://godbolt.org/z/5sn1dzhcE) sometimes surprises folks is why I prefer to not use mutable, but I'm fine to use here it and move on given that it's not a problem in this case.